xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/handler/packager/DefaultXWikiDocumentMerger.java

```diff
@@ -309,6 +313,15 @@ private XWikiDocument askDocumentToSave(XWikiDocument currentDocument, XWikiDocu
             }
         }
 
+        if (mergedDocument != null) {
+            Set<Right> rights = new HashSet<>(mergedDocument.getRequiredRights().getRights());
+            if (currentDocument != null) {
+                rights.addAll(currentDocument.getRequiredRights().getRights());
```

Comment:
Not sure to understand this logic? If it's merged already why adding those? Else if it's not merged why don't you consider the next document?

Reply:
Merge moved to `org.xwiki.store.merge.internal.DefaultMergeManager#mergeDocument`